File: ADIVINHAR PALAVRAS/c#/AdivinharPalavras/Principal/Dicionario.cs
using System;
using System.Collections.Generic;
using System.Text;


public class Dicionario
{

    #region Atributos e Propriedades
    private List<Palavra> palavras;

    #endregion

    #region Construtores
    public Dicionario()
    {
        palavras = new List<Palavra>();
    }
    #endregion

    #region Métodos Length, Add, Get e Remove
    public int Length()
    {
        return palavras.Count;
    }

    public void Add(Palavra novaPalavra)
    {
        palavras.Add(novaPalavra);
    }

    public Palavra Get(int indice)
    {
        return palavras[indice];
    }

    public Palavra GetRandom()
    {
        Random num = new Random();
        int indice = num.Next(0, Length());
        return Get(indice);
    }

    public Palavra[] GetRandomArray(int length)
    {
        Palavra[] palavrasTemp = new Palavra[length];

        for (int cont = 0; cont < length; cont++)
        {
            palavrasTemp[cont] = GetRandom();
        }

        return palavrasTemp;
    }

    public void Remove(int indice)
    {
        palavras.RemoveAt(indice);
    }

    public void Remove(Palavra palavraProcurada)
    {
        palavras.Remove(palavraProcurada);
    }
    #endregion
}

File: ADIVINHAR PALAVRAS/java/AdivinharPalavras/src/projeto/Dicionario.java
package projeto;

import java.util.ArrayList;

/**
 * @author Tiago Penha Pedroso
 * Date: 05/05/2018, 21:32:41
 */

public class Dicionario {
    
    //Atributos da classe
    public ArrayList<Palavra> palavras;
    
    //Atributos Auxiliares
    //private int indicePalavraAtual;
    
    public Dicionario(){
        palavras = new ArrayList<>();
    }
    
    public int length(){
        return palavras.size();
    }
    
    public void add(Palavra novaPalavra){
        palavras.add(novaPalavra);
    }
    
    public Palavra get(int indicePalavra){
        return palavras.get(indicePalavra);
    }
    
    public Palavra getRandom(){
        int indice = (int)(Math.random() * length());
        return get(indice);
    }
    
    public Palavra[] getRandomArray(int qtdPalavras){
        Palavra[] palavras = new Palavra[qtdPalavras];
        
        for (int cont = 0; cont < qtdPalavras; cont++) {
            palavras[cont] = getRandom();
        }
        
        return palavras;
    }
    
    public void remove(int indicePalavra){
        palavras.remove(indicePalavra);
    }
    
    public void remove(Palavra palavraProcurada){
        palavras.remove(palavraProcurada);
    }
    
}